### PR TITLE
Fixed `log_prob` for negative correlation in `SineBivariateVonMises`

### PR DESCRIPTION
**Pull request by OlaRonning**
Removes bias on correlation in `SBvM` normalization and fixes `NaN` in normalization for negative correlation (`corr < 0.`).  

From #1511.

### CODE REVIEW

File: numpyro/distributions/directional.py [hunk 1]
```diff
@@ -400,7 +400,7 @@ def norm_const(self):
 
         fs = (
             lbinoms.reshape(-1, 1)
-            + 2 * m * jnp.log(corr)
+            + m * jnp.log(corr**2)
```
Comment (fehiepsi) [inline numpyro/distributions/directional.py:403]: Nice catch! I guess you can clip by tiny here in case users set corr=0?

File: numpyro/distributions/directional.py [hunk 2]
```diff
@@ -400,7 +400,7 @@ def norm_const(self):
...
             - m * jnp.log(4 * jnp.prod(conc, axis=-1))
         )
         fs += log_I1(49, conc, terms=51).sum(-1)
```
Comment (fehiepsi) [inline numpyro/distributions/directional.py:406]: I'm not sure that I follow but it seems to me that norm_const does not have batch dimensions after fs.max() operator below. Is it intended?

Comment (fehiepsi) [reply numpyro/distributions/directional.py:406]: Probably you can remove mfs term because logsumexp does it under the hood.

File: numpyro/distributions/directional.py (hunk 1 above)
Comment (OlaRonning) [reply numpyro/distributions/directional.py:403]: Sure, that makes sense.

File: numpyro/distributions/directional.py (hunk 2 above)
Comment (OlaRonning) [reply numpyro/distributions/directional.py:405]: I see. I'll take out the `mfs` term.

File: numpyro/distributions/directional.py [hunk 3]
```diff
@@ -398,14 +398,12 @@ def norm_const(self):
         den = special.gammaln(m + 1.0)
         lbinoms = num - 2 * den
 
-        fs = (
-            lbinoms.reshape(-1, 1)
-            + 2 * m * jnp.log(corr)
-            - m * jnp.log(4 * jnp.prod(conc, axis=-1))
+        fs = lbinoms.reshape(-1, 1) + m * (
+            jnp.log(jnp.clip(corr**2, a_min=jnp.finfo(float).tiny))
```
Comment (fehiepsi) [inline numpyro/distributions/directional.py:402]: The float finfo is too small for single precision, how about `jnp.finfo(jnp.result_type(float)).tiny`?

**Review by fehiepsi — DISMISSED** [review]
Thanks, Ola! Just a small comment on the floating point.

**Review by fehiepsi — APPROVED** [review]
